sources/api/storewolf/src/main.rs

```diff
@@ -186,11 +187,48 @@ fn parse_metadata_toml(md_toml_val: toml::Value) -> Result<Vec<model::Metadata>>
             // A table means there is more processing to do. Add the current
             // key and value to the Vec to be processed further.
             toml::Value::Table(table) => {
-                for (key, val) in table {
-                    trace!("Found table for key '{}'", &key);
-                    let mut path = path.clone();
-                    path.push(key.to_string());
-                    to_process.push((path, val));
+                // This is special case to handle setting generator metadata as table.
+                // Following are the keys for setting-generator as table:
+                // "command": "command",
+                // "strength": "weak",
+                // "skip-if-populated": true
+                if table.contains_key("command") && table.contains_key("strength") {
+                    ensure!(
+                        !path.is_empty(),
```

Comment:
There's a bug where if you write a generator like so, it won't be properly represented as a single object:

```toml
[metadata.settings.foo.setting-generator]
command = "foo"
skip-if-populated = false
```

Instead of inspecting the inner keys of the `table` to determine its type, we should instead inspect the keyname. In other words, if we encounter the key name `setting-generator`, then we should assume that a nested `Table` or `String` are both acceptable as-is, and don't need to be "recursively" broken down.